lib/handlebars/compiler/javascript-compiler.js

```diff
@@ -165,12 +165,10 @@ JavaScriptCompiler.prototype = {
 
       let { programs, decorators } = this.context;
       for (i = 0, l = programs.length; i < l; i++) {
-        if (programs[i]) {
```

Comment:
Thinking out loud: since the `programs`-object isn't mutated inside the for-loop, I guess it's safe to omit this check.

Reply:
Yes, this check was due to the current implementation creating gaps in the array. The coverage tool complained, presumably because it always enters there so it's redundant, and I had to remove this to remain at 100% coverage.